apps/actions-dependencies-updater/src/utils.mts

```diff
@@ -0,0 +1,166 @@
+import { existsSync } from "node:fs";
+import { readdir, readFile } from "node:fs/promises";
+import { join } from "node:path";
+
+import * as log from "./logger.mjs";
+import { Action, WorkflowByName } from "./workflows.mjs";
+
+/**
+ * Get an environment variable or throw an error if it's not set
+ * @param envVariableName The name of the environment variable
+ * @returns The value of the environment variable
+ */
+export function getEnvironmentVariableOrThrow(envVariableName: string): string {
+  const variable = process.env[envVariableName];
+  if (!variable) {
+    throw Error(`${envVariableName} not set`);
+  }
+  return variable;
+}
+
+/**
+ * From a list of tags, "guess" the latest version.
+ * @param tags A list of tags, without the prefixed with "refs/tags/"
+ * @returns The latest version
+ */
+export function guessLatestVersion(
+  tags: string[],
+  repo?: string,
+  repoPath?: string,
+) {
+  let versions = tags.map((tag) => parseTagToVersion(tag));
+
+  // support for the .github monorepo
+  if (repo === ".github" && repoPath) {
+    let actionName = repoPath.split("/").pop();
+    if (actionName === "signed-commits")
+      actionName = "changesets-signed-commits"; // hacky fix
+    log.debug(
+      `Guessing latest version for monorepo action: ${actionName} (${repo})`,
+    );
+    if (actionName) {
+      versions = versions.filter((v) => v.prefix === actionName);
+    }
+  }
+
+  // Sort the versions by comparing major, minor, and patch numbers
+  versions.sort((a, b) => {
+    if (a.major !== b.major) return a.major - b.major;
+    if (a.minor !== b.minor) return a.minor - b.minor;
+    return a.patch - b.patch;
+  });
```

Comment:
We should use semver sort instead https://www.npmjs.com/package/semver

Reply:
There's a lot of nice utilities in that package. `semver.coerce` would've saved me a good amount of time lmao

Reply:
Using - 50cc1960ceed8ebe36a118d48cc38d39c5827fb2